test_xpartition.py

```diff
@@ -145,3 +145,49 @@ def test_dataset_mappable_write(tmpdir, ds, ranks):
 
     result = xr.open_zarr(store)
     xr.testing.assert_identical(result, ds)
+
+
+@pytest.mark.parametrize("has_coord", [True, False])
+def test_PartitionMapper_integration(tmpdir, has_coord):
+    def func(ds):
+        return ds.rename(z="new_name").assign_attrs(dataset_attr="fun")
+
+    ds = xr.Dataset({"z": (["x", "y"], np.ones((5, 10)), {"an": "attr"})},).chunk(
+        {"x": 2}
+    )
+    if has_coord:
+        ds = ds.assign_coords(x=range(5))
+
+    store = str(tmpdir)
+    mapper = ds.z.partition.map(store, ranks=3, dims=["x"], func=func, data=ds)
+    for rank in mapper:
+        mapper.write(rank)
+
+    written = xr.open_zarr(store)
+    xr.testing.assert_identical(func(ds), written)
+
+
+def test_partition_partition():
+    # Partitions have two qualitities which we test using the a data-array that
+    # has all unique values
+    ds = xr.Dataset(
+        {"z": (["x", "y"], np.arange(50).reshape((5, 10)), {"an": "attr"})},
```

Comment:
nit: I think the attribute does not add anything to this particular test (it's useful in the one above).
```suggestion
        {"z": (["x", "y"], np.arange(50).reshape((5, 10)))},
```